### PR TITLE
Alert app proto v2

### PR DESCRIPTION
**Pull request by regit**
Update of #2799 implementing app_proto or partial flow entry in alert events.

PR builds:
- PR regit: https://buildbot.openinfosecfoundation.org/builders/regit/builds/302
- PR regit-pcap: https://buildbot.openinfosecfoundation.org/builders/regit-pcap/builds/86

### CODE REVIEW

**Comment by inliniac** [comment]
<pre>
  CC       output.o
  CC       output-file.o
In file included from output.c:49:0:
output-json-flow.h:28:27: error: unknown type name ‘json_t’
 void JsonAddFlow(Flow *f, json_t *js, json_t *hjs);
                           ^
output-json-flow.h:28:39: error: unknown type name ‘json_t’
 void JsonAddFlow(Flow *f, json_t *js, json_t *hjs);
                                       ^
  CC       output-filedata.o
make[3]: *** [Makefile:1662: output.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/cygdrive/c/projects/suricata/src'
make[2]: *** [Makefile:1110: all] Error 2
make[2]: Leaving directory '/cygdrive/c/projects/suricata/src'
make[1]: *** [Makefile:477: all-recursive] Error 1
make[1]: Leaving directory '/cygdrive/c/projects/suricata'
make: *** [Makefile:404: all] Error 2
</pre>

File: src/output-json-alert.c [hunk 1]
```diff
@@ -85,8 +86,9 @@
 #define LOG_JSON_DNP3           BIT_U16(8)
 #define LOG_JSON_VARS           BIT_U16(9)
 #define LOG_JSON_APP_LAYER      BIT_U16(10)
+#define LOG_JSON_FLOW           BIT_U16(11)
 
-#define LOG_JSON_APP_LAYER_ALL  (LOG_JSON_APP_LAYER|LOG_JSON_HTTP|LOG_JSON_TLS|LOG_JSON_SSH|LOG_JSON_SMTP|LOG_JSON_DNP3)
+#define LOG_JSON_APP_LAYER_ALL  (LOG_JSON_APP_LAYER|LOG_JSON_HTTP|LOG_JSON_TLS|LOG_JSON_SSH|LOG_JSON_SMTP|LOG_JSON_DNP3|LOG_JSON_FLOW)
```
Comment (inliniac) [inline src/output-json-alert.c]: I think LOG_JSON_APP_LAYER_ALL should only include app layer protocols

Comment (inliniac) [inline src/output-json-alert.c]: perhaps it makes sense to rename it to LOG_JSON_METADATA or similar

Comment (regit) [inline src/output-json-alert.c]: Should we rename the option in config in that case ?

Comment (inliniac) [inline src/output-json-alert.c]: good idea. Probably just 'metadata'

Comment (regit) [inline src/output-json-alert.c]: You wanna include vars in the set by default ?

Comment (inliniac) [inline src/output-json-alert.c]: Yes

**Pull request closed without merging**